Bucketeer/Sources/Public/BKTConfig.swift

```diff
@@ -11,60 +11,146 @@ public struct BKTConfig {
         if pollingInterval < Constant.MINIMUM_POLLING_INTERVAL_MILLIS {
-            logger?.warn(message: "pollingInterval: \(pollingInterval) is set but must be above \(Constant.MINIMUM_POLLING_INTERVAL_MILLIS)")
+            builder.logger?.warn(message: "pollingInterval: \(pollingInterval) is set but must be above \(Constant.MINIMUM_POLLING_INTERVAL_MILLIS)")
             pollingInterval = Constant.MINIMUM_POLLING_INTERVAL_MILLIS
         }
-        var backgroundPollingInterval = backgroundPollingInterval
+        var backgroundPollingInterval : Int64 = builder.backgroundPollingInterval ?? Constant.MINIMUM_BACKGROUND_POLLING_INTERVAL_MILLIS
         if backgroundPollingInterval < Constant.MINIMUM_BACKGROUND_POLLING_INTERVAL_MILLIS {
-            logger?.warn(message: "backgroundPollingInterval: \(backgroundPollingInterval) is set but must be above \(Constant.MINIMUM_BACKGROUND_POLLING_INTERVAL_MILLIS)")
+            builder.logger?.warn(message: "backgroundPollingInterval: \(backgroundPollingInterval) is set but must be above \(Constant.MINIMUM_BACKGROUND_POLLING_INTERVAL_MILLIS)")
             backgroundPollingInterval = Constant.MINIMUM_BACKGROUND_POLLING_INTERVAL_MILLIS
         }
-        var eventsFlushInterval = eventsFlushInterval
+        var eventsFlushInterval: Int64 = builder.eventsFlushInterval ?? Constant.DEFAULT_FLUSH_INTERVAL_MILLIS
         if eventsFlushInterval < Constant.MINIMUM_FLUSH_INTERVAL_MILLIS {
-            logger?.warn(message: "eventsFlushInterval: \(eventsFlushInterval) is set but must be above \(Constant.MINIMUM_FLUSH_INTERVAL_MILLIS)")
+            builder.logger?.warn(message: "eventsFlushInterval: \(eventsFlushInterval) is set but must be above \(Constant.MINIMUM_FLUSH_INTERVAL_MILLIS)")
             eventsFlushInterval = Constant.DEFAULT_FLUSH_INTERVAL_MILLIS
         }
 
-        self = BKTConfig(
-            apiKey: apiKey,
-            apiEndpoint: apiEndpointURL,
-            featureTag: featureTag,
-            eventsFlushInterval: eventsFlushInterval,
-            eventsMaxQueueSize: eventsMaxQueueSize,
-            pollingInterval: pollingInterval,
-            backgroundPollingInterval: backgroundPollingInterval,
-            sdkVersion: Version.current,
-            appVersion: appVersion,
-            logger: logger
-        )
+        let eventsMaxQueueSize = builder.eventsMaxQueueSize ?? Constant.DEFAULT_MAX_QUEUE_SIZE
+
+        self.init(apiKey: apiKeyForSDK,
+                  apiEndpoint: apiEndpointURL,
+                  featureTag: featureTag,
+                  eventsFlushInterval: eventsFlushInterval,
+                  eventsMaxQueueSize: eventsMaxQueueSize,
+                  pollingInterval: pollingInterval,
+                  backgroundPollingInterval: backgroundPollingInterval,
+                  sdkVersion: builder.sdkVersion,
+                  appVersion: appVersion,
+                  logger: builder.logger)
+    }
+
+    public class Builder {
```

Comment:
`Builder` as an inner class.
The benefit here, every time the consumer wants to create a `BKConfig` by entering `BKConfig.`
The IDE will show a suggestion to use the `Builder` 

<img width="928" alt="Screenshot 2023-07-01 at 15 34 08" src="https://github.com/bucketeer-io/ios-client-sdk/assets/2597710/425b3744-99ea-4913-a7b4-f7fa6e3c6a9e">